dashboard/lib/logic/task_grid_filter.dart

```diff
@@ -151,8 +161,13 @@ class TaskGridFilter extends FilterPropertySource {
       'showLinux': 'linux',
       'showAndroid': 'android',
     };
+
+    // Replacing mac_ios by ios only, so that filtering out mac doesn't remove ios.
+    // Same for *_android
+    String compareStr =
```

Comment:
but won't this keep showing mac_ios if ios is unchecked but mac is still checked. Same for linux_android and windows_android.

Reply:
I may have the logic wrong (I didn't run it) but `LinkedHashMap` is ordered so I think `Mac_ios task` would match `ios` and return the value of `showiOS` or false (true might be more correct, I don't know if they are always set in `_allProperties`). ``Mac task` would not match `ios` but would match `mac` and return the value of of `showMac`.

You don't need to use this exact sample, but I found the method hard to understand.
1. That `showMac` etc are properties in `_allProperties`
2. The replacement/regex seems unnecessary (since you could instead process them in order)
3. That matching the task string is needed to return false, it seemed like it should return true because it matches. I get it now `Mac_ios task` should return the value of `showiOS` but I had to stare at it awhile to understand that.

Reply:
> I don't think we can get rid of the replace()

Hm, why not?  If it matches `ios` like `mac_ios` then we use the value of `showiOS` and return.  If it doesn't match `ios` but does match `mac` then we use `showMac`.  Same logic for mac/android Linux/android, etc.  As long as you check `android` and `ios` first I don't think you need the replace or regex.

Reply:
Now I understood :D sorry for that didn't grasp the whole thing. Don't know why I kept on thinking that it will always go through the whole map. Thanks. I'll update